src/Symfony/Component/Translation/Translatable.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation;
+
+/**
+ * @author Nate Wiebe <nate@northern.co>
+ */
+final class Translatable
+{
+    private $messageId;
+
+    private $parameters;
+
+    private $domain;
+
+    public function __construct(string $messageId, array $parameters = [], string $domain = 'messages')
+    {
+        $this->messageId = $messageId;
```

Comment:
what about throwing a LogicException here, if either $messageId or $domain is an empty string

Reply:
I'm good with adding an exception if the domain is empty, but I think throwing one if the message is empty forces the user to add more if empty checks. If the message is empty, it will already be handled by the translator and return an empty string. So I think we're good there.

Reply:
Actually, still on the fence about this. Depending on how this gets used, not sure if it makes sense to throw an exception on user error. For example, let's say a translator is responsible for adding translations and it's done through the web frontend. If they have the ability to set the domain and potentially make it empty or invalid, we would see message ids. Which if the ids are just English, it's not very intrusive. If we throw an exception instead, it's likely going to take out the whole page/site depending on where the issue is. I think it's probably best to leave it for now.

Reply:
fair :+1: keeping the responsibility in the translator is fine.